src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectOptionsProcessor.cs

```diff
@@ -33,12 +30,10 @@ internal class ProjectSystemProjectOptionsProcessor : IDisposable
     private Checksum? _commandLineChecksum;
 
     /// <summary>
-    /// To save space in the managed heap, we dump the entire command-line string to our
-    /// temp-storage-service. This is helpful as compiler command-lines can grow extremely large
-    /// (especially in cases with many references).
+    /// To save space in the managed heap, we only cache the command line if we have an
+    /// effective ruleset.
```

Comment:
```suggestion
    /// ruleset.
```

Not that "effective" is a bit wrong, but it makes me think the logic here is fancier than it actually is.